qcodes/tests/test_deprecate.py

```diff
@@ -27,8 +26,47 @@ def _add_one(x):
     @deprecate(reason='this function is for private use only')
     def add_one(x):
         return _add_one(x)
-    with warnings.catch_warnings(record=True) as w:
+    with assert_deprecated(
+            'The function <add_one> is deprecated, because '
+            'this function is for private use only.'):
         assert add_one(1) == _add_one(1)
-        assert (str(w[-1].message) ==
-                'The function <add_one> is deprecated, because '
-                'this function is for private use only.')
+
+
+def test_deprecated_context_manager():
+    with _catch_deprecation_warnings() as ws:
+        issue_deprecation_warning('something')
+        issue_deprecation_warning('something more')
+        warnings.warn('Some other warning')
+    assert len(ws) == 2
+    with warnings.catch_warnings(record=True) as ws:
+        issue_deprecation_warning('something')
+        warnings.warn('Some other warning')
+    assert len(ws) == 2
+
+
+def test_deprecated_class():
+
+    @deprecate(reason='this is a test')
+    class C:
+        a = 'pristine'
+
+        def method(self):
+            self.a = 'last called by method'
+
```

Comment:
for the sake of sanity-checking, shall classes with explicit `__init__`, `@property` also be tested?

Reply:
good idea!

Reply:
So property does not work. I suggest we document it and move on to more pressing issues.

Reply:
agree! (please keep the failing test ;) )